Containers/talk/start.sh

```diff
@@ -16,43 +19,47 @@ elif [ -z "$INTERNAL_SECRET" ]; then
 fi
 
 set -x
-IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short)"
+if [ -n "$(dig nextcloud-aio-talk A +short | grep -E "^[0-9.]+$" | sort | head -n1)" ]; then
+    IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short | grep -E "^[0-9.]+$" | sort | head -n1)"
+fi
+if [ -n "$(dig nextcloud-aio-talk AAAA +short | grep -E "^[0-9a-fA-F:]+$" | sort | head -n1)" ]; then
+    IPv6_ADDRESS_TALK="$(dig nextcloud-aio-talk AAAA +short | grep -E "^[0-9a-fA-F:]+$" | sort | head -n1)"
+fi
+
+if [ -n "$(dig "$NC_DOMAIN" A +short | grep -E "^[0-9.]+$" | sort | head -n1)" ]; then
+    IPv4_ADDRESS_NC="$(dig "$NC_DOMAIN" A +short | grep -E "^[0-9.]+$" | sort | head -n1)"
+fi
+if [ -n "$(dig "$NC_DOMAIN" AAAA +short | grep -E "^[0-9a-fA-F:]+$" | sort | head -n1)" ]; then
+    IPv6_ADDRESS_NC="$(dig "$NC_DOMAIN" AAAA +short | grep -E "^[0-9a-fA-F:]+$" | sort | head -n1)"
+fi
```

Comment:
another question, what will happen to local instances?

Reply:
you mean offering TURN in a local network? TURN would not be needed in local (internal) environments as direct p2p is mostly used. However, if needed, the respective CIDRs should be whitelisted then if blacklist `recommended` is used. 
The advertised relay IP address can be a local address. It is only important that the user group which should be able to reach the TURN server cat actually reach it.

Reply:
Yes, what about cases when NC_DOMAIN is configured via a local dns server to point to a local domain locally? I suppose this would probably run into problems for people then that connect from the outside?

Reply:
yeah, hence he suggestion to introduce a variable `TURN_DOMAIN` which defaults to `NC_DOMAIN`, see above in the other section.

Reply:
Would the stun server be able to figure out the correct public ip-address in that case?

Reply:
Ah I see.

I guess it makes most sense then to remove the logic around nc_domain again and keep the automatic discovery because it should get the correct public ip in most cases that we want to support.

Reply:
Does Nextcloud offer a public STUN server for everybody to use? You could use that as the default. The STUN client integrated in eturnal only supports UDP. The STUN server of course all transports: UDP, TCP, TLS.

Reply:
> Does Nextcloud offer a public STUN server for everybody to use?

yes, Nextcloud does. See below :)

Reply:
ENV STUN_SERVICE="stun.nextcloud.com 443"

Reply:
https://github.com/nextcloud/all-in-one/pull/2855#discussion_r1241233958